### PR TITLE
Enable builds for Kyma Metrics Collector

### PR DESCRIPTION
**Pull request by sayanh**
**Description**

Changes proposed in this pull request:

- Enable builds for Kyma metrics collector

### CODE REVIEW

**Review by k15r — DISMISSED** [review]

File: templates/config.yaml [hunk 1]
```diff
@@ -329,6 +329,13 @@ templates:
           path: components/metris
           since: "1.14"
           optional: false
+      - to: ../prow/jobs/control-plane/components/kyma-metrics-collector/kyma-metrics-collector-generic.yaml
+        values:
+          <<: *cluster_default
+          <<: *control_plane_generic_component
+          path: components/kyma-metrics-collector
+          since: "1.21"
```
Comment (adamwalach) [inline templates/config.yaml:337]: We already started with release 1.21,  so I think this value should be changed to "1.22"

File: development/tools/go.sum [hunk 2]
```diff
@@ -819,6 +819,8 @@ golang.org/x/lint v0.0.0-20191125180803-fdd1cda4f05f/go.mod h1:5qLYkcX4OjUUV8bRu
 golang.org/x/lint v0.0.0-20200130185559-910be7a94367/go.mod h1:3xt1FjdF8hUf6vQPIChWIBhFzV8gjjsPE/fR3IyQdNY=
 golang.org/x/lint v0.0.0-20200302205851-738671d3881b h1:Wh+f8QHJXR411sJR8/vRBTZ7YapZaRvUcLFFJhusH0k=
 golang.org/x/lint v0.0.0-20200302205851-738671d3881b/go.mod h1:3xt1FjdF8hUf6vQPIChWIBhFzV8gjjsPE/fR3IyQdNY=
+golang.org/x/lint v0.0.0-20201208152925-83fdc39ff7b5 h1:2M3HP5CCK1Si9FQhwnzYhXdG6DXeebvUHFpre8QvbyI=
```
Comment (adamwalach) [inline development/tools/go.sum:822]: Changes in go.sum and go.mod are nod needed. We can have a separate PR with dependency updates

Comment (sayanh) [reply development/tools/go.sum]: Reverted

File: templates/config.yaml (hunk 1 above)
Comment (sayanh) [reply templates/config.yaml]: Fixed

**Review by adamwalach — APPROVED** [review]

**Pull request merged**